themes/default/content/blog/announcing-v6-of-the-pulumi-aws-classic-provider/index.md

```diff
@@ -0,0 +1,60 @@
+---
+title: "Announcing v6 of the Pulumi AWS Classic Provider"
+date: 2023-08-08T09:00:00-04:00
+draft: false
+meta_desc: "Announcing v6 of the Pulumi AWS Classic Provider, including SDK size optimization, TF plugin framework support, new resources, and more."
+meta_image: meta.png
+authors:
+    - mikhail-shilkov
+    - monica-rodriguez
+tags:
+    - aws
+    - providers
+
+---
+
+We are excited to announce v6.0 of the Pulumi AWS Classic provider. The AWS Classic provider is the most heavily used provider across the entire Pulumi ecosystem, and offers access to the full surface area of the upstream Terraform AWS Provider from within Pulumi projects in all supported Pulumi languages. The v6.0 release brings a substantial set of fixes and improvements to the provider, including a number of breaking changes as part of the major version release.
+
+This blog post walks you through the list of notable changes in the new major version.
+
+<!--more-->
+
+### SDK Size Optimization
+
+The AWS Classic provider contains more than 1,200 resources with over 12,000 properties between them. As a result, it’s expected that the SDK size of the provider has grown to be fairly large. With that in mind, we invested in some key areas to reduce the size of generated files. Depending on the runtime, the SDK is now up to 90% smaller than it used to be in v5 of the provider!
+
+![aws_sdk](aws_sdk.png)
+
+One particular optimization is related to `wafv2` resources, that were particularly difficult to use in Pulumi AWS v5 (issue [one](https://github.com/pulumi/pulumi-aws/issues/2276), [two](https://github.com/pulumi/pulumi-aws/issues/1117[), [three](https://github.com/pulumi/pulumi-aws/issues/2250)). The new release includes refactored `wafv2` types to recursive shapes, which enabled us to remove a lot of excess generated code.
```

Comment:
Are there words missing from the last sentence in this paragraph? I don't understand "includes refactored `wafv2` types to recursive shapes".

Reply:
@mikhailshilkov Please take a look here.

Reply:
@scottslowe This is clearer in the [migration guide](https://github.com/pulumi/registry/pull/2941/files) as the examples are given. I'll update the text here to match.

```diff
-pulumi_aws.wafv2.RuleGroupRuleStatementAndStatementStatementGeoMatchStatementArgs
+pulumi_aws.wafv2.RuleGroupRuleStatementArgs
```

Reply:
I took a look at the migration guide and I think I have a better idea of what you're trying to say. Can I suggest this reworded text, then?

> The new release includes a refactored `wafv2` module which uses recursive types. Using recursive types allowed us to remove a lot of excess generated code.